scripts/shellcheck.sh

```diff
@@ -0,0 +1,253 @@
+        exit 0
+    fi
+}
+
+view_report() {
+    # view non-empty shellcheck report, includes info, warnings, errors
+    if [[ "$VIEW" == "True" ]]; then
+        echo; echo -e "----------------------- ${BLUE}Start of Shellcheck report${RST_COLOR} -----------------------"; echo
+        cat "$SHELLCHECK_REPORT"
+        echo; echo -e "------------------------ ${BLUE}End of Shellcheck report${RST_COLOR} ------------------------"; echo
+    fi
+}
+
+determine_pass_fail() {
+    # detect if fatal errors are in shellcheck report
+    echo -e "Checking for errors in shellcheck report with severity level '${CYAN}${SEVERITY}${RST_COLOR}'"; echo
+    echo -e "Excluding error codes: '${YELLOW}${EXCLUDE_CODES}${RST_COLOR}'"; echo
+
+    # if severity not set, set it to error as default
+    if [[ -z "$SEVERITY" ]]; then
+        SEVERITY="error"
+    fi
+
+    # if any messages of severity level or more strict are present, use for grepping report
+    case "$SEVERITY" in
+        "info")    REGEX_SEVERITY=("info" "warning" "error") ;;
+        "warning") REGEX_SEVERITY=("warning" "error") ;;
+        "error")   REGEX_SEVERITY=("error") ;;
+    esac
+
+    TOTAL_ERRORS=
+    for SEV_LEVEL in "${REGEX_SEVERITY[@]}"; do
+        # get count of errors of severity level
+        ERROR_COUNT=$(grep -Ec "\(${SEV_LEVEL}\):" "$SHELLCHECK_REPORT")
+        TOTAL_ERRORS=$(( TOTAL_ERRORS + ERROR_COUNT ))
+        if [[ "$ERROR_COUNT" -gt 0 ]]; then
+            echo -e "Found ${RED}${ERROR_COUNT}${RST_COLOR} violations of '${CYAN}${SEV_LEVEL}${RST_COLOR}' severity level"; echo
+        fi
+    done
+
+    # if no errors found, then shellcheck passed
+    if [[ "$TOTAL_ERRORS" -eq 0 ]]; then
+        echo -e "${GREEN}[PASS]${RST_COLOR} Shellcheck did not detect violations scanning with severity level '${CYAN}${SEVERITY}${RST_COLOR}'"
+        echo; echo -e "${GREEN}Shellcheck passed.${RST_COLOR} See report: ${SHELLCHECK_REPORT}"; echo
+        exit 0
+    else
+        echo -e "${RED}[FAIL]${RST_COLOR} Shellcheck found unexcused violations scanning with severity level '${CYAN}${SEVERITY}${RST_COLOR}'"
+        echo; echo -e "${RED}Shellcheck failed.${RST_COLOR} See report: ${SHELLCHECK_REPORT}"; echo
+        exit 1
+    fi
+}
+
+main() {
+    # exit if shellcheck is not installed
+    check_shellcheck_install
+    # parse command line arguments
+    parse_cli_args "$@"
+    # run shellcheck in parallel on all tracked shell scripts
+    get_num_cores
+    # checking status of this run will give failure if info/warning/error is found by default
+    run_shellcheck
+    # check if shellcheck report was created
+    check_report
+    # display report contents if view set in cli
+    view_report
+    # determine status by parsing shellcheck report to see if any messages
+    # of the severity level or more strict are present to determine failure (true errors)
+    determine_pass_fail
+}
+
+main "$@"
```

Comment:
```suggestion
check_shellcheck_install
parse_cli_args "$@"
get_num_cores
run_shellcheck
check_report
view_report
determine_pass_fail
```

If even all of these make it in, there's no benefit to including a `main` function. And the comments here are all redundant as the functions being called are actually well-named (though, the control flow here should probably be extracted out of the internal functions so that, for example, `view_report` only even gets called if the user asked for it).

Reply:
ACK, thanks about the function naming.
I put this inside `check_report()`
```console
    if [[ "$VIEW" == "True" ]]; then
        view_report
    fi
```
and removed that logic from `view_report()` so it will shows the report if called.